src/main/scala/viper/gobra/frontend/info/implementation/typing/ExprTyping.scala

```diff
@@ -333,13 +342,60 @@ trait ExprTyping extends BaseTyping { this: TypeInfoImpl =>
             case _ => assignableTo.errors(l, UNTYPED_INT_CONST)(n) ++ assignableTo.errors(r, UNTYPED_INT_CONST)(n)
           }
           case (_: PAdd, StringT, StringT) => noMessages
-          case (_: PAdd | _: PSub | _: PMul | _: PMod | _: PDiv, l, r) =>
-            if (l == PermissionT || r == PermissionT || getTypeFromCtxt(n.asInstanceOf[PNumExpression]).contains(PermissionT)) {
+          case (_: PAdd | _: PSub | _: PMul | _: PMod | _: PDiv | _: PBitAnd | _: PBitOr | _: PBitXor | _: PBitClear, l, r) =>
+            if (l == PermissionT || r == PermissionT || getTypeFromCtxt(n).contains(PermissionT)) {
```

Comment:
Do we effectively allow bitwise operations with permissions on lhs and rhs?

Reply:
This was initially my idea but then I opted to not do it - I don't think it makes sense to have bitwise operators in permissions because they do not have a standard bit representation. However, I forgot to change this.